### DISCUSSION TITLE
Convert ODK transformer response to support web bot

### DISCUSSION THREAD

**Discussion by Nik720** [Samagra-Development/SamagraX-community#40] [Request for new features]
@ChakshuGautam

Right now ODK transformer is combining title and choices and send them back as plain text, which is fine for the WhatsApp scenario. 
For the web Tara bot, we have to show choices as a button, and using plain text format it is difficult to identify title and choices. so need to convert ODK transformer response as below structure so any frontend application can render it.

`{
     title: "Tell me which board you belong to",
     choices: [
         {key: 1, text:"1. CBSE"},
        { key:2, text:"2. State Board"}
     ]
 }
`

cc @vinukumar-vs @aayushkedia95

**Comment by vinukumar-vs** [comment]
@ChakshuGautam As we discussed, if we enhace the XMessage spec styles as below to handle the above 
```
<text>Title of the message</text>
<buttons>
    <button id="{choice_1_name}">Choice 1 label</button>
    <button id="{choice_2_name}">Choice 2 label</button>
    <button id="{choice_3_name}">Choice 3 label</button>
</buttons>
```

When there is no choice for the user selected option then
```
<text>Title of the message</text>
```

The above XMessage format is just for reference. Feel free to change.

**Comment by Nik720** [comment]
@ChakshuGautam

1) In the key-based approach suppose I have specified options like` name: course | lablel: 2 Course` and the user enters `course` then I am not getting a response. 
If I type `2` or `2 Course` exactly the same label then I am getting a response. Is it right? If yes then you have to give a numeric button id here `<button id="{choice_1_name}">Choice 1 label</button>`

2) Is there any update on how can we start a state-level workflow from the global ODK form? you were supposed to test it on samagra sandbox and show us a demo?

3) Suppose I want to initiate my bot with multiple starting messages like "Hi Tara", "Hi", "Hello", "Tara" is it possible? Do I need to create multiple bots for this?

cc @vinukumar-vs @aayushkedia95

**Reply by Nik720** [reply]
@ChakshuGautam @aayushkedia95 Is there any update on the above?

cc @vinukumar-vs

**Reply by aayushkedia95** [reply]
@Nik720, 
1. This is correct. However, we get your requirement. We will update the code and get back to you by tomorrow. 
2. We will deploy the updated code by tomorrow, and post that it should work fine. 
3. Currently, it is not possible. We can think of creating a separate issue to add this to the development.  

cc - @vinukumar-vs @ChakshuGautam

**Reply by Nik720** [reply]
@aayushkedia95 

> 2\. We will deploy the updated code by tomorrow, and post that it should work fine.

Is deployment is done?

**Reply by Nik720** [reply]
Is there any update?  @ChakshuGautam @ChakshuGautam

**Reply by aayushkedia95** [reply]
Hi @Nik720, there are certain build related issues that we are trying to resolve. It will be done by EoD. I will update you.

**Reply by Nik720** [reply]
Is there any update? @ChakshuGautam @aayushkedia95

**Reply by aayushkedia95** [reply]
Hi @Nik720, let's discuss this on our catchup call today.

**Reply by Nik720** [reply]
> 2\. Is there any update on how can we start a state-level workflow from the global ODK form? you were supposed to test it on samagra sandbox and show us a demo?

Hi @aayushkedia95 

When can we have a demo call for the above one?

cc - @vinukumar-vs @ChakshuGautam

**Reply by aayushkedia95** [reply]
Hi @Nik720, have scheduled a call for tomorrow morning 10am.  

cc - @vinukumar-vs @ChakshuGautam

**Reply by Nik720** [reply]
Hi @aayushkedia95  @ChakshuGautam

http://143.110.255.220:10021/odk/updateAll 
this link is not working, I want to restart the ODK server 

cc - @vinukumar-vs

**Reply by aayushkedia95** [reply]
Try this - http://143.110.255.220:19091/odk/updateAll
